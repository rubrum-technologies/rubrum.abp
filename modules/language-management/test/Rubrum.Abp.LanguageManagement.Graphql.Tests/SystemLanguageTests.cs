using CookieCrumble;
using HotChocolate.Types.Relay;
using Shouldly;
using Xunit;

namespace Rubrum.Abp.LanguageManagement;

public class SystemLanguageTests : LanguageManagementGraphqlTestBase
{
    private readonly IIdSerializer _idSerializer;

    public SystemLanguageTests()
    {
        _idSerializer = GetRequiredService<IIdSerializer>();
    }

    [Fact]
    public async Task FetchById()
    {
        var id = _idSerializer.Serialize(null, SystemLanguageConstants.TypeName, "ru");

        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  systemLanguageById(id: "{{id}}") {
                      id
                      code
                      name
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task Fetch()
    {
        var id = _idSerializer.Serialize(null, SystemLanguageConstants.TypeName, "en");

        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              query {
                  systemLanguage(where: { id: { eq: "{{id}}" } }) {
                      id
                      code
                      name
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchList()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            """
            query {
                systemLanguages {
                    nodes {
                      id
                      code
                      name
                    }
                    pageInfo {
                      endCursor
                      hasNextPage
                      hasPreviousPage
                      startCursor
                    }
                }
            }
            """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchAny()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            """
            query {
                systemLanguagesAny(where: { name: { eq: "Русский" } })
            }
            """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task FetchCount()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            """
            query {
                systemLanguagesCount
            }
            """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task Create()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            """
            mutation {
                createSystemLanguage (input: { code: "test", name: "Test" }) {
                    systemLanguage {
                        id
                        code
                        name
                    }
                    errors {
                        ... on Error {
                            message
                        }
                    }
                }
            }
            """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task Update()
    {
        var id = _idSerializer.Serialize(null, SystemLanguageConstants.TypeName, "zh");
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              mutation {
                  updateSystemLanguage (input: { id: "{{id}}", name: "CH" }) {
                      systemLanguage {
                          id
                          code
                          name
                      }
                      errors {
                          ... on Error {
                              message
                          }
                      }
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task Delete()
    {
        var id = _idSerializer.Serialize(null, SystemLanguageConstants.TypeName, "gu");
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            $$"""
              mutation {
                  deleteSystemLanguage (input: { id: "{{id}}" }) {
                      systemLanguage {
                          id
                          code
                          name
                      }
                      errors {
                          ... on Error {
                              message
                          }
                      }
                  }
              }
              """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }
}
