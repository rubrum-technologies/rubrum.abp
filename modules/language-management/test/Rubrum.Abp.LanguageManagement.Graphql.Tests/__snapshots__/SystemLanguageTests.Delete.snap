{
  "data": {
    "deleteSystemLanguage": {
      "systemLanguage": {
        "id": "U3lzdGVtTGFuZ3VhZ2UKZGd1",
        "code": "gu",
        "name": "हिन्दी"
      },
      "errors": null
    }
  }
}